src/ios/AppsFlyerPlugin.m

```diff
@@ -166,6 +191,65 @@ - (void)setCurrencyCode:(CDVInvokedUrlCommand*)command
     NSString* currencyId = [command.arguments objectAtIndex:0];
     [AppsFlyerLib shared].currencyCode = currencyId;
 }
+
+/**
+*   Sets new currency code. currencyId: ISO 4217 Currency Codes.
+*/
+- (void)setConsentData:(CDVInvokedUrlCommand*)command
+{
+    if ([command.arguments count] == 0) {
+        return;
+    }
+    NSDictionary *consentDataMap = (NSDictionary*)[command.arguments objectAtIndex: 0];
+
+    BOOL isUserSubjectToGDPR = NO;
+    BOOL hasConsentForDataUsage = NO;
+    BOOL hasConsentForAdsPersonalization = NO;
+
+    id isUserSubjectToGDPRValue = nil;
+    id hasConsentForDataUsageValue = nil;
+    id hasConsentForAdsPersonalizationValue = nil;
+
+    isUserSubjectToGDPRValue = [consentDataMap objectForKey:@"isUserSubjectToGDPR"];
+    if ([isUserSubjectToGDPRValue isKindOfClass:[NSNumber class]]) {
+       isUserSubjectToGDPR = [(NSNumber*)isUserSubjectToGDPRValue boolValue];
```

Comment:
i don't think that the casting to `NSNumber` here is necessary because you already checked that the class is NSNumber

Reply:
fixed